payu/experiment.py

```diff
@@ -995,6 +980,56 @@ def sweep(self, hard_sweep=False):
         if os.path.islink(self.work_sym_path):
             print('Removing symlink {0}'.format(self.work_sym_path))
             os.remove(self.work_sym_path)
+    
+    def prune_restarts(self, from_n_restart=0, to_n_restart=None):
+        """Returns a list of restart directories that can be pruned"""
+        restart_freq = self.config.get('restart_freq', 5)
+        restart_history = self.config.get('restart_history', default_restart_history)
+
+        # All restarts directories
+        restarts = [d for d in os.listdir(self.archive_path)
+                                if d.startswith('restart')]
+        # Sort restarts based on counter - in increasing date order
+        restarts.sort(key=lambda d: int(d.lstrip('restart')))
+
+        # Note: from_n_restart and to_end_restart could be useful for inspecting only the more recent restarts
+        if to_n_restart is None:
+            # Keep restart_history n restarts 
+            to_n_restart = -restart_history
+        restarts = restarts[from_n_restart:to_n_restart] 
+
+        restarts_to_prune = []
+        if self.repeat_run:
+            # TODO: Previous logic was to prune all restarts if self.repeat_run - is that still the case?
+            restarts_to_prune = [os.path.join(self.archive_path, restart) for restart in restarts]
```

Comment:
I was going to use MOM6 for this, but as you pointed out this is not supported by the MOM6 driver. Having said which, they both use FMS, so MOM6 has an `ocean_solo.res` file.

So ... if this was an FMS driver it would work with any model that sub-classes `Fms`, which includes both `mom5` and `mom6`.

In which case I might rescind my comment about `proleptic_gregorian`.

Reply:
Ok, so move the `mom`'s `get_restart_datetime()` to the fms model driver? The other model that subclasses of `Fms` is GOLD, so would it be worth checking if there is an `ocean_solo.res` exists and if not raise an not implemented error?